monai/data/dataset.py

```diff
@@ -323,9 +315,6 @@ def _pre_transform(self, item_transformed):
             random transform object
 
         """
-        if not isinstance(self.transform, Compose):
```

Comment:
Since both these datasets inherit from `Dataset` and the check is now moved to `Dataset`, this point in the code would always receive a `Compose` object, making this check redundant.

Reply:
When considering it along with the previous question, `PersistentDataset` must ensure that the transform is always not None. However, it will always convert to a compose in the base class, even with an empty transform, results in behavior that is inconsistent with the previous implementation.

Reply:
Correct me if I'm wrong but I wonder if these checks were redundant in the original class as well? The lines in the constructor that this PR removes were converting `transform` to a `Compose` object if it wasn't already, so that is the behaviour that now is inherited. In these methods we're checking that `self.transform` is a `Compose` object, but wouldn't that always be the case in the existing code and in this PR? Am I missing some situation where this check would prevent some condition?

Reply:
No, you are right. The previous implementation is also redundant. Mark as resolved.